app/src/main/java/dev/andrylat/raqimbek/bankingutils/CardValidator.java

```diff
@@ -0,0 +1,124 @@
+package dev.andrylat.raqimbek.bankingutils;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.stream.IntStream;
+
+public class CardValidator {
+    public CardValidationInfo checkCardNumber(String input) {
+        var errors = new ArrayList<String>();
+        var isValid = false;
+        
+        if (!containsOnlyDigits(input)) {
+            errors.add("Card Number must contain only digits");
+        }
+        
+        for (PaymentSystem paymentSystem : PaymentSystem.values()) {
+            if (!hasValidPrefix(input, paymentSystem)) {
+                errors.add("Payment System can't be determined");
+            }
+
+            if (!hasValidLength(input, paymentSystem)) {
```

Comment:
In the task I sent you link where only "16" symbols card numbers are supported. Why did you start to implement so complex logic without need?